src/components/TransactionOperation.tsx

```diff
@@ -1,60 +1,156 @@
-import { OperationRecord, TransactionOperation } from 'js-kinesis-sdk'
-import { startCase } from 'lodash'
+import {
+  CreateAccountOperationRecord,
+  OperationRecord,
+  PaymentOperationRecord,
+} from 'js-kinesis-sdk'
 import * as React from 'react'
 
+import { AddressView } from '@components/AddressView'
 import { HorizontalLabelledField } from '@components/LabelledField'
-import { TransactionOperationView } from '@types'
+import { Currency, TransactionOperationView } from '@types'
 
 export interface Props {
   transactionWithOperation: TransactionOperationView
+  currency: Currency
 }
 
-// keyof BaseOperationRecord
-const baseOperationRecordKeys = [
-  'id',
-  '_links',
-  'effects',
-  'paging_token',
-  'precedes',
-  'self',
-  'succeeds',
-  'transaction',
-  'type',
-  'type_i',
-]
-
-export const renderOperationRecords = (operation: OperationRecord | TransactionOperation) => {
-  const entries = Object.entries(operation).filter(
-    ([key, value]) => !baseOperationRecordKeys.includes(key) && ['string'].includes(typeof value),
-  )
-  return entries.map(([key, value]) => (
-    <HorizontalLabelledField key={key} label={startCase(key)} value={value} isCompact={true} />
-  ))
+const isTransfer = (
+  op: OperationRecord,
+): op is PaymentOperationRecord | CreateAccountOperationRecord =>
+  ['create_account', 'payment'].includes(op.type)
+
+const getAddress = (t: TransactionOperationView) => {
+  switch (t.operation.type) {
+    case 'create_account':
+      return t.isIncoming ? t.operation.funder : t.operation.account
+    case 'payment':
+      return t.isIncoming ? t.operation.from : t.operation.to
+    default:
+      return t.source
+  }
 }
 
-export const TransactionCard: React.SFC<Props> = ({ transactionWithOperation: t }) => (
-  <article className="message is-small is-transparent">
-    <div className="message-header">
-      <p>
-        {!t.isIncoming && (
-          <span className="icon has-text-success">
-            <i className="fas fa-arrow-down" />
-          </span>
-        )}
-        {t.isIncoming && (
-          <span className="icon has-text-danger">
-            <i className="fas fa-arrow-up" />
+const getAmount = (t: TransactionOperationView) => {
+  let amount = 0
+  switch (t.operation.type) {
+    case 'create_account':
+      amount = Number(t.operation.starting_balance)
+      break
+    case 'payment':
+      amount = Number(t.operation.amount)
+      break
+    default:
+      return 'Other'
+  }
+  return t.isIncoming ? amount : amount + Number(t.fee)
+}
+
+interface StateProps {
+  moreInfoIsHidden: boolean
+  toggleMoreInfo: () => any
+}
+const TransactionCard: React.SFC<Props & StateProps> = ({
+  transactionWithOperation: t,
+  currency,
+  moreInfoIsHidden,
+  toggleMoreInfo,
+}) => (
+  <article className="level">
```

Comment:
This render is a bit of a beast, and hard to read. Want to make a few smaller logical SFCs?